packages/web/src/components/pages/WebsiteResults.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect } from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import { Error as ErrorLayout, Website } from 'components/layouts';
+import { trackCustomEvent } from '../../services/analytics';
+import {
+  useAppDispatch,
+  useAppSelector,
+  applyFilters,
+  getWebsite,
+  websiteResultsSelectors as selectors,
+} from '../../store';
+import { FiltersState } from '../layouts/Filters/Filters';
+
+export function WebsiteResultsPage() {
+  const { hostname } = useParams();
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const { webpages, vulnerabilities } = useAppSelector(selectors.default);
+  const packagesFiltered = useAppSelector(selectors.packagesSortedAndFiltered);
+  const isProtected = useAppSelector(selectors.isProtected);
+  const isPending = useAppSelector(selectors.isPending);
+  const isLoading = useAppSelector(selectors.isLoading);
+  const isError = useAppSelector(selectors.isFailed);
+  const isInvalidResult = useAppSelector(selectors.isInvalid);
+  const setFilters = (filters: FiltersState) => dispatch(applyFilters(filters));
+
+  useEffect(() => {
+    if (hostname && !isLoading && isPending) {
+      dispatch(getWebsite(hostname));
+    }
+  });
+
+  // TODO: properly handle history/routing
+  useEffect(() => {
+    if (!hostname || isError) {
+      navigate('/', { replace: true });
+    }
+  });
+
+  if (isProtected) {
+    trackCustomEvent('HostnamePage', 'SiteProtected');
```

Comment:
Tracking events on each re-render seems like a bad practice

Reply:
There will be no further rerenders in this condition branch. Also, see my previous comment about tracking events in redux - probably we'll move it from here.

Reply:
Please leave a TODO